src/pint/models/fdjump.py

```diff
@@ -0,0 +1,207 @@
+    tempo2-compatible.
+
+    Caveat
+    ------
+    `FDJUMP`s have two indices: the polynomial/FD/prefix index and the system/mask
+    index. i.e., they have properties of both `maskParameter`s such as `JUMP`s and
+    `prefixParameters` such as `FD`s. There is currently no elegant way in PINT
+    to implement such parameters due to the way parameter indexing is implemented;
+    there is no way to distinguish between mask and prefix indices.
+
+    Hence, they are implemented here as `maskParameter`s as a stopgap measure.
+    This means that there must be an upper limit for the FD indices. This is controlled
+    using the ``pint.models.fdjump.fdjump_max_index`` variable, and is 20 by default.
+    Note that this is strictly a limitation of the implementation and not a property
+    of `FDJUMP`s themselves.
+
+    `FDJUMP`s appear in tempo2-format par files as `FDJUMPp`, where p is the FD index.
+    The mask index is not explicitly mentioned in par files similar to `JUMP`s.
+    PINT understands both `FDJUMPp` and `FDpJUMP` as the same parameter in par files,
+    but the internal representation is always `FDpJUMPq`, where q is the mask index.
+
+    PINT understands 'q' as the mask parameter just fine, but the identification of 'p'
+    as the prefix parameter is done in a hacky way.
+
+    This implementation may be overhauled in the future.
+
+    Parameters supported:
+
+    .. paramtable::
+        :class: pint.models.fdjump.FDJump
+    """
+
+    register = True
+    category = "fdjump"
+
+    def __init__(self):
+        super().__init__()
+
+        # Matches "FDpJUMPq" where p and q are integers.
+        self.param_regex = re.compile("^FD(\\d+)JUMP(\\d+)")
+
+        self.add_param(
+            boolParameter(
+                name="FDJUMPLOG",
+                value=False,
+                description="Whether to use log-frequency for computing FDJUMPs.",
+            )
+        )
+        for j in range(1, fdjump_max_index + 1):
+            self.add_param(
+                maskParameter(
+                    name=f"FD{j}JUMP",
+                    units="second",
+                    description=f"System-dependent FD parameter of polynomial index {j}",
+                )
+            )
+
+        self.delay_funcs_component += [self.fdjump_delay]
+
+    def setup(self):
+        super().setup()
+
+        self.fdjumps = [
+            mask_par
+            for mask_par in self.get_params_of_type("maskParameter")
+            if self.param_regex.match(mask_par)
+        ]
+
+        for fdj in self.fdjumps:
+            # prevents duplicates from being added to phase_deriv_funcs
+            if fdj in self.deriv_funcs.keys():
+                del self.deriv_funcs[fdj]
+            self.register_deriv_funcs(self.d_delay_d_FDJUMP, fdj)
+
+    def get_fd_index(self, par):
+        """Extract the FD index from an FDJUMP parameter name. In a parameter name
+        "FDpJUMPq", p is the FD/prefix index and q is the mask index.
```

Comment:
Use consistent style here and in the top-level docstring.

Reply:
changed the top level docstring to use double quotes